src/main/java/Main.java

```diff
@@ -8,6 +9,79 @@ public static void main(String[] args) {
     }
 
     public void run(Scanner scanner) {
+        ArrayList<Integer> numbers = inputNumbers(scanner);
+        printReversedNumbers(numbers);
+        printResultOfAdding(numbers);
+        try {
+            printTheSmallestNumber(numbers);
+            printTheBiggestNumber(numbers);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Nie dodano elementów do listy, lista jest pusta.");
+        }
```

Comment:
Wyjątki staraj się używać w "wyjątkowych sytuacjach". Trochę trudno to opisać, ale w tym przypadku możesz ifem sprawdzić, czy lista nie jest pusta i w ten sposób uniknąć korzystania z wyjątków. Przerób to proszę na wersję z ifem. To samo tyczy się łapania np NullPointerException. Zamiast tego sprawdza się czy coś nie jest nullem.